x-pack/plugins/security_solution/common/endpoint/types.ts

```diff
@@ -379,6 +395,7 @@ export interface LegacyEndpointEvent {
   event?: {
     action?: string;
     type?: string;
+    category?: string | string[];
```

Comment:
Is this coming from the sensor? It seems odd for it to be a string or a string array. Any code working with it will have to handle both the string and string[] case. If the type were `string[]`, then code would only have to handle that case.

Reply:
Yeah this structure models the event coming from the endgame sensor. So currently, I believe we're only sending a single value for the category field from the endgame sensor, but ECS defines the field as an array. So I'm trying to guard against always assuming it's a single string.

For the endpoint we already check to see if it is a string or array: https://github.com/elastic/kibana/blob/master/x-pack/plugins/security_solution/public/resolver/models/process_event.ts#L53